modules/rabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,294 @@
+	})
+}
+
+// WithConfigErlang overwrites the default RabbitMQ configuration file with the supplied one.
+// The "configFilePath" parameter holds the path to the file to use (in sysctl format, don't forget empty line in the end of file)
+// and it will check if the file has the ".config" extension.
+func WithConfigErlang(confFilePath string) testcontainers.CustomizeRequestOption {
+	return withConfig(confFilePath, defaultCustomConfigErlangPath, func(s string) bool {
+		return strings.HasSuffix(s, ".config")
+	})
+}
+
+func withConfig(hostPath string, containerPath string, validateFn func(string) bool) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		if !validateFn(hostPath) {
+			return
+		}
+
+		req.Env["RABBITMQ_CONFIG_FILE"] = containerPath
+
+		req.Files = append(req.Files, testcontainers.ContainerFile{
+			HostFilePath:      hostPath,
+			ContainerFilePath: containerPath,
+			FileMode:          0o644,
+		})
+	}
+}
+
+// WithExchange declares the exchange on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithExchange(e Exchange) testcontainers.CustomizeRequestOption {
+	return withExecutable(e)
+}
+
+// WithOperatorPolicy declares the operator policy on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithOperatorPolicy(op OperatorPolicy) testcontainers.CustomizeRequestOption {
+	return withExecutable(op)
+}
+
+// WithParameter declares the parameter on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithParameter(p Parameter) testcontainers.CustomizeRequestOption {
+	return withExecutable(p)
+}
+
+// WithPermission declares the permission on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithPermission(p Permission) testcontainers.CustomizeRequestOption {
+	return withExecutable(p)
+}
+
+// WithPolicy declares the policy on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithPolicy(p Policy) testcontainers.CustomizeRequestOption {
+	return withExecutable(p)
+}
+
+// WithQueue declares the queue on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithQueue(q Queue) testcontainers.CustomizeRequestOption {
+	return withExecutable(q)
+}
+
+// WithVirtualHost declares the virtual host on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithVirtualHost(v VirtualHost) testcontainers.CustomizeRequestOption {
+	return withExecutable(v)
+}
+
+// WithVirtualHostLimit declares the virtual host limit on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithVirtualHostLimit(l VirtualHostLimit) testcontainers.CustomizeRequestOption {
+	return withExecutable(l)
+}
+
+// WithVirWithusertualHost declares the user on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithUser(u User) testcontainers.CustomizeRequestOption {
+	return withExecutable(u)
+}
```

Comment:
In general, all of those fn helpers can be achieved by `container.Exec` but are coupling the container implementation to the tool cli and that's not intended. tc-java rabbitmq implementation should deprecate and remove those helper methods in future versions. At least, that was the last agreement on what the testcontainers implementation should provide.

Reply:
Ok, I see. But if we want to release a module in the same version as Java, then this implementation is fine, right?

We can evolve the module once we a new design for providing such capabilities is available. Or is the intention to remove them all without replacement?

Reply:
> Or is the intention to remove them all without replacement?

That is currently the preferred case, yes. Like so often with modules, it comes down to the trade-off between _integration-surface_, _api-surface_ and _usability_. I think our current preference lies with minimizing the _integration-surface_.

Hence, I would prefer to remove the methods here in Go and merge the PR, independent from the methods still being present in Java.

Reply:
I'm going to proceed and remove the majority of the WithXXX functions, and keep the examples, but calling container.Exec instead. An approach could be keeping the types in the test code, for reference and not to maintain them as production code, and use them in the examples to produce the commands to be executed into the container